custom_components/tahoma/config_flow.py

```diff
@@ -49,3 +49,20 @@ async def async_step_user(self, user_input=None):
         return self.async_show_form(
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
+
+    async def async_step_import(self, import_info):
+        """Handle YAML configuration."""
+
+        if import_info is not None:
+            username = import_info.get(CONF_USERNAME)
+            password = import_info.get(CONF_PASSWORD)
+
+            await self.async_set_unique_id(username)
+            self._abort_if_unique_id_configured()
+
+            async with TahomaClient(username, password) as client:
+                try:
+                    await client.login()
+                    return self.async_create_entry(title=username, data=import_info)
+                except Exception as exception:  # pylint: disable=broad-except
+                    _LOGGER.exception(exception)
```

Comment:
Some duplicate code with `async_step_user` you can create a method to avoid this.

Reply:
@tetienne any suggestion? There are quite some parts that are specific for the method. I wasn't sure for a way that would actually improve readability.

Reply:
From the beginning to almost the end of the method this is the same code, so can simply cut the code and paste it in another method.